File: Android/Android-Master/10-Animaciones/animDemo/app/src/main/java/com/animaciones/MainActivity.java
package com.animaciones;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.drawable.AnimationDrawable;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private TextView tvFadeAnim;    //1

    private AnimationDrawable animationDrawable;    //2
    private ImageView ivDrawable;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvFadeAnim = findViewById(R.id.tvFadeAnim);     //1

        ivDrawable = findViewById(R.id.ivDrawable);     //2
        ivDrawable.setBackgroundResource(R.drawable.anim_check);
        animationDrawable = (AnimationDrawable) ivDrawable.getBackground();

    }

    public void startFadeAnimation(View view) {
        if(tvFadeAnim.getVisibility() == View.VISIBLE) tvFadeAnim.setVisibility(View.GONE);
        else tvFadeAnim.setVisibility(View.VISIBLE);
    }   //1

    public void startDrawableAnimation(View view) {
        animationDrawable.start();
    }    //2
}

File: Android/Android-jetpack/1-DataBinding/StarterProject/StarterProject/BindingDemo2/app/src/main/java/com/anushka/bindingdemo2/MainActivity.kt
package com.anushka.bindingdemo2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import androidx.databinding.DataBindingUtil
import com.anushka.bindingdemo2.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding   //TODO:Variable global de binding
    //private lateinit var button: Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main)   //TODO: Referenciamos layout con binding
//        setContentView(R.layout.activity_main)
        binding.apply {
            controlButton.setOnClickListener(){
                startOrStopProgressBar()
            }
        }
        //button = findViewById(R.id.control_button)
//        button.setOnClickListener {
//            startOrStopProgressBar()
//        }
    }
      private fun startOrStopProgressBar(){
          binding.apply {
              if(progressBar.visibility == View.GONE){
                  progressBar.visibility = View.VISIBLE
                  controlButton.text = "Stop"
              }else{
                  progressBar.visibility = View.GONE
                  controlButton.text = "Start"
              }
          }
      }
//    private fun startOrStopProgressBar() {
//        val progressBar = findViewById<ProgressBar>(R.id.progressBar)
//        if (progressBar.visibility == View.GONE) {
//            progressBar.visibility = View.VISIBLE
//            button.text = "Stop"
//        } else {
//            progressBar.visibility = View.GONE
//            button.text = "Start"
//        }
//    }
}

